### PR TITLE
[core] add linter

### PR DESCRIPTION
**Pull request by neversaynever0502**
LIQ-917

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,5 +49,13 @@
...
     "react-dom": "./core/ui/node_modules/react-dom/profiling",
     "scheduler/tracing": "./core/ui/node_modules/scheduler/tracing-profiling"
+  },
+  "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^4.14.0",
+    "@typescript-eslint/parser": "^4.14.0",
+    "eslint": "^7.18.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react-hooks": "^4.4.0"
```
Comment (haVincy) [inline package.json:59]: Should use `devDependencies` ?

File: package.json [hunk 2]
```diff
@@ -19,7 +19,8 @@
     "clean:sdk": "rm -rf core/sdk/node_modules && rm -rf core/sdk/dist",
     "clean:types": "rm -rf core/types/node_modules && rm -rf core/types/dist",
     "clean:ui": "rm -rf core/ui/node_modules && rm -rf core/ui/dist",
-    "clean:all": "yarn clean:root && yarn clean:sdk && yarn clean:ui && yarn clean:types"
+    "clean:all": "yarn clean:root && yarn clean:sdk && yarn clean:ui && yarn clean:types",
+    "lint": "eslint \"core/{sdk,cli,types,ui}/**/*.ts\""
```
Comment (haVincy) [inline package.json:23]: If we are adding this in yarn script, do we need to enable pre-push hook and CI to check it? @RustySol

**Comment by haVincy** [comment]
If we gonna make this linter, I think we better do the lint fix on all codes in the same PR as coverage.

File: package.json (hunk 2 above)
Comment (RustySol) [reply package.json:23]: Yes that will be the plan. @neversaynever0502  pushed this up for our early review.

File: package.json (hunk 1 above)
Comment (neversaynever0502) [reply package.json]: Yes, I have changed it.

File: example/eth/ShopConfig.tsx [hunk 3]
```diff
@@ -83,7 +83,7 @@ export const ShopConfig: React.FC<ShopConfigProps> = ({ setCandyForm, candyForm
 
   useEffect(() => {
     if (candyForm) onUpdateFormState(candyForm);
-  }, [candyForm]);
+  }, [candyForm, onUpdateFormState]);
```
Comment (RustySol) [inline example/eth/ShopConfig.tsx:86]: this change seems not related to linting

**Review by RustySol — APPROVED** [review]

File: example/eth/ShopConfig.tsx (hunk 3 above)
Comment (RustySol) [reply example/eth/ShopConfig.tsx:86]: turns out it's related to `react-hooks/exhaustive-deps`

**Review by vanminh1701 — APPROVED** [review]

**Comment by vanminh1701** [comment]
nice config for projects 👍
